File: src/main/java/org/joel3112/componentbuilder/ui/settingsComponent/components/NameComponent.java
package org.joel3112.componentbuilder.ui.settingsComponent.components;

import com.intellij.ui.components.JBTextField;
import com.intellij.util.ui.FormBuilder;
import org.joel3112.componentbuilder.settings.data.MutableState;
import org.joel3112.componentbuilder.ui.settingsComponent.Component;

import javax.swing.*;

public class NameComponent implements Component, MutableState<String> {
    private final JBTextField textField;

    public NameComponent(String name) {
        textField = new JBTextField(name);
    }

    @Override
    public void addToBuilder(FormBuilder builder) {
        JPanel panel = new JPanel();
        panel.add(new JLabel("Name:"));
        panel.add(textField);
        builder.addLabeledComponent(panel, new JLabel());
    }

    @Override
    public String getState() {
        return textField.getText();
    }

    @Override
    public void setState(String state) {
        textField.setText(state);
    }
}

File: src/main/kotlin/org/joel3112/componentbuilder/settings/ui/settingsComponent/components/NameComponent.kt
package org.joel3112.componentbuilder.settings.ui.settingsComponent.components

import com.intellij.ui.components.JBTextField
import com.intellij.util.ui.FormBuilder
import org.joel3112.componentbuilder.settings.data.MutableState
import org.joel3112.componentbuilder.settings.ui.settingsComponent.Component
import javax.swing.JLabel
import javax.swing.JPanel

class NameComponent(name: String?) : Component, MutableState<String> {
    private val textField = JBTextField(name)

    override fun addToBuilder(formBuilder: FormBuilder) {
        val panel = JPanel()
        panel.add(JLabel("Name:"))
        panel.add(textField)
        formBuilder.addLabeledComponent(panel, JLabel())
    }

    override var state: String
        get() = textField.text
        set(state) {
            textField.text = state
        }
}
